src/dali_backend.cc

```diff
@@ -396,19 +398,25 @@ TRITONBACKEND_ModelInstanceExecute(
     const uint32_t request_count)
 {
   TRITONSERVER_Error* error = nullptr;  // success
+  uint64_t exec_start_ns, exec_end_ns,
+      batch_exec_start_ns = -1, batch_exec_end_ns = -1,
+      batch_compute_start_ns = -1, batch_compute_end_ns = -1;
+  int total_batch_size = 0;
   DaliModelInstance* instance_state;
   RETURN_IF_ERROR(TRITONBACKEND_ModelInstanceState(
       instance, reinterpret_cast<void**>(&instance_state)));
   std::vector<TRITONBACKEND_Request*> requests(reqs, reqs + request_count);
   std::vector<TRITONBACKEND_Response*> responses(request_count);
 
   for (size_t i = 0; i < responses.size(); i++) {
+    exec_start_ns = detail::capture_time();
```

Comment:
You should move the decl of error = nullptr to here. You need to set 'error' = nullptr for each request. In general need to think about what happens if one or more requests have error.

Reply:
Done